src/EnergyPlus/SolarShading.cc

```diff
@@ -459,7 +459,7 @@ void GetShadowingInput(EnergyPlusData &state)
             state.dataIPShortCut->cAlphaArgs(aNum) = "PolygonClipping";
 #else
             auto error_callback = [](const int messageType, const std::string &message, void *contextPtr) {
-                auto *state = (EnergyPlusData *)contextPtr;
+                auto *state = reinterpret_cast<EnergyPlusData *>(contextPtr);
```

Comment:
I have nothing against some uses of C-style casts, but I do against some uses. Possible should have added an `assert(state != nullptr);` here.